src/installer/corehost/cli/hostpolicy/deps_resolver.cpp

```diff
@@ -876,6 +876,19 @@ bool deps_resolver_t::resolve_probe_dirs(
         }
     }
 
+    // If this is a single-file app, add the app's dir to the native search directories.
+    if (bundle::info_t::is_single_file_bundle() && !is_resources)
+    {
+        auto bundle = bundle::runner_t::app();
+        add_unique_path(asset_type, bundle->base_path(), &items, output, &non_serviced, core_servicing);
+
+        // If native components are self-extracted, add the extraction path too.
+        if (bundle->needs_native_libraries_extraction())
+        {
+            add_unique_path(asset_type, bundle->extraction_path(), &items, output, &non_serviced, core_servicing);
```

Comment:
Should this check that the extraction path is non-empty?

Reply:
I just changed the if condition to check that the directory exists, I don't see any reason why we should also check if it's empty.